service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -132,11 +135,14 @@ private List<DatasetResponse> collectDatasets(StorageSystem system) {
     return createDatasetResponses(roleMap, system);
   }
 
-  public DatasetsListResponse listDatasets() {
+  public DatasetsListResponse listDatasets() throws InterruptedException {
```

Comment:
I think we should keep the current convention that these APIs don't throw checked exceptions. This could catch `Interrupted` and rethrow another exception instead. (Although note that sonar will want you to interrupt the current thread when you catch `InterruptedException`.)